File: agent-python/agent/supply.py
from typing import Literal

from .position import Position

SupplyKind = Literal[
    "PRIMARY_ARMOR",
    "PREMIUM_ARMOR",
    "S686",
    "M16",
    "AWM",
    "VECTOR",
    "BANDAGE",
    "FIRST_AID",
    "BULLET",
    "GRENADE",
]


class Supply:
    def __init__(self, kind: SupplyKind, count: int, position: Position[float]):
        self.kind = kind
        self.count = count
        self.position = position

    def __str__(self) -> str:
        return f"Supply{{kind: {self.kind}, count: {self.count}, position: {self.position}}}"

File: agent-cpp/src/agent/supply.cc
#include "supply.h"

#include <fmt/format.h>

#include <magic_enum.hpp>
#include <string>

namespace thuai7_agent {

auto format_as(SupplyKind object) -> std::string {
  return std::string(magic_enum::enum_name(object));
}

auto format_as(Supply const& object) -> std::string {
  return fmt::format("Supply{{kind: {}, count: {}, position: {}}}", object.kind,
                     object.count, object.position);
}

}  // namespace thuai7_agent
